### DISCUSSION TITLE
Plastic Case modification

### DISCUSSION THREAD

**Discussion by electricityforprogress** [joeycastillo/Sensor-Watch#33] [General]
I was wondering if additional space inside the case could be free'd up by modifying the plastic case.  I was looking around for a 3D model (must be out there some where), and I wanted to ask if anyone had a lead on a 3d model for the F91W?

-sam

**Comment by electricityforprogress** [comment]
I found this model, but I'm not sure if it is truely accurate. https://grabcad.com/library/casio-f-91w-watch

I'll post up my designs once I cad up a replacement insert.

**Comment by joeycastillo** [comment]
I'm going to try printing the watch case from that model, it looks like it might be pretty close!

As for a mod, the [Data Runner](https://n-o-d-e.net/datarunner.html) is a cool F-91W modification project by NODE that offered a couple of interesting design files (search "Download Source Files" on that page), including a backplate extender. 

![Screen Shot 2022-01-12 at 5 12 12 PM](https://user-images.githubusercontent.com/1765152/149231181-66be6793-adaa-4bdf-a4e3-b539657d8e6b.png)

This mod adds additional space on the back side of the watch between the battery enclosure and the backplate. There are some notes on that page about it and in the video, namely that you need to bend some metal contacts further out to make the piezo buzzer work. Anyway I see no reason this mod wouldn't also work with Sensor Watch. 

I should print this one too, come to think of it :)

**Reply by electricityforprogress** [reply]
Thank you very much Joeyan extender was exactly what I was hoping to craft up!  Good luck tonight in the Adafruit Show-and-tell!

**Comment by electricityforprogress** [comment]
Ha, so after trying to print the watch parts on my Prusa Mini, it is obvious that an SLA printer would do a much better job!  The spacer seemed to print just fine ;)

**Comment by joeycastillo** [comment]
So this is interesting: the model in the GrabCad link is staggeringly close to being a perfect fit. With one tweak, it's perfect.

![both](https://user-images.githubusercontent.com/1765152/149412755-e07fd544-28a8-47e5-88c5-d1786c28dda1.jpg)

The one tweak is that the cavity for the watch movement was 2mm too short. The width is fine, just the height needs to be adjusted. My colleague @rohansingh made the necessary adjustments, and you can [grab the STL here](https://github.com/joeycastillo/Sensor-Watch/files/7866174/plastik-dis.stl.zip)!

**Reply by rohansingh** [reply]
Here's the STEP file in case anyone wants to make further modifications:
[casio-v2.step.xz.zip](https://github.com/joeycastillo/Sensor-Watch/files/7866267/casio-v2.step.xz.zip)


Apologies for stupid double-compressed package, required for GitHub upload.

**Reply by electricityforprogress** [reply]
Your print came out much nicer than mine, I tried using a very fine layer height.  I'll try with the modified STL, thanks @rohansingh for tweaking the file!

**Comment by Ethan-Ming** [comment]
![IMG_0918](https://user-images.githubusercontent.com/36059080/170808093-f44c48b0-5200-460f-a4fa-27d25fd985ed.JPG)

I'm Making a **Pop Up Secret-compartment** mod to the f91w, I'm thinking of a pop-lit LED for a start

But I'm having trouble recreating the TF card style push-push mechanism working on a 3d printed parts

Do any good hearted smart engineers care to share some helping ideas?

**Reply by electricityforprogress** [reply]
I recommend using an actual metal card popsocket, and 3D printing a shell which can hold it. You're not going to get a good spring action on a 3D model in that size, and the little aluminum bent popsockets are extremely tiny on their own. Then you could 3D print something in the shape of an SD card to hold your LED.

**Reply by Ethan-Ming** [reply]
Any specific popsocket model you have in mind?
or perhaps can point me to what type of parts I should be sourcing 
thanks

**Comment by electricityforprogress** [comment]
uxcell SD Memory Card Socket Long Body 11 Pin PCB Mount Connector 20pcs https://www.amazon.com/dp/B07KY7QR9G/ref=cm_sw_r_apan_i_9AK68BY9FS7PJ9H397BV
I would think that any available SD card socket would work, I'm sure there's lots of different types out there. But I haven't necessarily sampled lots of them. Need to make sure that it's push-push rather than pop-up. Here is a link to the evil 'zon

**Comment by Ethan-Ming** [comment]
I did have brought some of theses , the issue with them is the travel
distance is too small if I just use them as it is without any modification
( which introduce  manufacturing difficulties at this size) , but thanks
for the suggestion anyways

**Comment by electricityforprogress** [comment]
I did some more research on 3d printed mechanisms. It seems like there are some push/push sprint loaded designs, but nothing any where near small enough to fit in the watch.  Would probably need sla printing for the detail resolution.   In the end most mechanisms are a spring, a moving platform, and two small notches to catch the platform in tension.  https://www.thingiverse.com/thing:4544557

**Comment by Ethan-Ming** [comment]
Thanks! with this I think I can remix and shrink it to fit into the watch
with SLA service, I update the progress here once I have a functioning
parts

**Reply by electricityforprogress** [reply]
I am thinking that something very simple, like the drawer in your image of parts, a single spring behind a drawer and a small tab which the drawer tucks behind to hold tension. ...  the best 'stash drawer' would probably just be a friction fit slide with a small tab to grab with a finger nail.

Good luck with your remix and I'll keep an eye out for another 'pop' print solution
